src/pyhf/pdf.py

```diff
@@ -222,12 +222,22 @@ def __init__(self, spec, **config_kwargs):
             spec, self.channel_nbins
         )
 
-        poiname = config_kwargs.get('poiname', 'mu')
+        poiname = config_kwargs.pop('poiname', 'mu')
+
+        # measurement_name is explicitly removed in cases where the model is created using poi_name, multiple
+        # measurements are there but the first one is taken or if measurement is defined as None in the argument
+        if 'measurement_name' in config_kwargs:
+            del config_kwargs['measurement_name']
+
         default_modifier_settings = {'normsys': {'interpcode': 'code1'}}
-        self.modifier_settings = (
-            config_kwargs.get('modifier_settings') or default_modifier_settings
+        self.modifier_settings = config_kwargs.pop(
+            'modifier_settings', default_modifier_settings
         )
 
+        assert (
+            not config_kwargs
+        ), f"""Unexpected keyword argument(s) '{"', '".join(config_kwargs.keys())}'"""
```

Comment:
It would be nice to get a test added to `test_pdf.py` that checks this behavior and the error type.